src/source/CppGenerator.scala

```diff
@@ -37,12 +37,12 @@ class CppGenerator(spec: Spec) extends Generator(spec) {
     var hppFwds = mutable.TreeSet[String]()
     var cpp = mutable.TreeSet[String]()
 
-    def find(ty: TypeRef) { find(ty.resolved) }
-    def find(tm: MExpr) {
-      tm.args.foreach(find)
-      find(tm.base)
+    def find(ty: TypeRef, dontIncludeRecordsAndEnums: Boolean) { find(ty.resolved, dontIncludeRecordsAndEnums) }
```

Comment:
The `dontInclude` is kinda confusing - how about `forwardDeclareRecordsAndEnums`? Name it for what it does, not for what it doesn't do.